src/lib.rs

```diff
@@ -2,4 +2,89 @@
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
 
-include!(concat!(env!("OUT_DIR"), "/cuda_driver_bindings.rs"));
+use std::ffi::CString;
+use std::sync::{Mutex, Once};
+use std::borrow::BorrowMut;
+
+use futures::{
+    executor::block_on,
+};
+
+use tarpc::{
+    client,
+    context, 
+    tokio_serde::formats::Json,
+};
+
+mod service;
+
+pub type CUresult = ::std::os::raw::c_uint;
+pub type cuuint32_t = u32;
+pub type cuuint64_t = u64;
+pub type CUdeviceptr_v2 = ::std::os::raw::c_ulonglong;
+pub type CUdeviceptr = CUdeviceptr_v2;
+pub type CUdevice_v1 = ::std::os::raw::c_int;
+pub type CUdevice = CUdevice_v1;
+
+static mut CUDA_CLIENT: Option<Mutex<service::CudaClient>> = None;
+static INIT: Once = Once::new();
+
+fn cudaClient<'a>() -> &'a Mutex<service::CudaClient> {
+    INIT.call_once(|| {
+        // Since this access is inside a call_once, before any other accesses, it is safe
+        unsafe {
+            let transport = block_on(tarpc::serde_transport::tcp::connect("[::1]:50055", Json::default)).unwrap();
```

Comment:
why port `50055`? I would like to keep accurate track of all of the ports we use, without having to audit the code base later.

Reply:
just a number.. have to choose something..